### PR TITLE
Integrate Google Maps API

### PR DESCRIPTION
**Pull request by nadinekuo**
- Basic Flutter application setup that allows you to navigate across pages using the bottom bar - a big portion of changed files relate to this oops
- Google Maps API integration in home screen with support for markers and drawing polygons, related files:
  - `lib/views/home.dart`
  - `lib/main.dart`
  - `lib/services/location_service.dart`
  - `pubspec.yaml`
  - See `apa/src/apa_app/README.md` for info on how to set up private key environment (for Google Maps e.g.)

image

The search bar can be used to quickly navigate to some city.
image

User can draw random polygons.

image

Emulated on tablet:
image

Note that the color scheme/logo/title of the app are to be modified still :) Also, the drawing of polygons is very basic right now, but we may perform validations on user taps still.

### CODE REVIEW

File: src/apa_app/lib/views/home.dart [hunk 1]
```diff
@@ -0,0 +1,182 @@
...
+  //
+  // void _setMarker(LatLng point) {
+  //   setState(() {
+  //     _markers.add(
+  //       Marker(
+  //           markerId: MarkerId('marker'),
+  //           position: point),
+  //     );
+  //   });
+  // }
```
Comment (paulmis) [inline src/apa_app/lib/views/home.dart:100]: Stray code

File: src/apa_app/lib/views/home.dart [hunk 2]
```diff
@@ -0,0 +1,182 @@
...
+    // Give each polygon a different id
+    _polygonIdCounter++;
+
+    _polygons.add(
+      Polygon(
+        polygonId: PolygonId(polygonIdVal),
+        points: polygonLatLngs,
+        strokeWidth: 3,
+        strokeColor: Colors.orange,
+        fillColor: Colors.transparent,
```
Comment (paulmis) [inline src/apa_app/lib/views/home.dart:81]: Regarding the colors, can I read somewhere about the color palette of the frontend?

File: src/apa_app/lib/main.dart [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
+      ),
+      initialRoute: '/',
+      routes: {
+        '/': (context) => const MainPage(),
+        '/load': (context) => Loading(),
+        '/home': (context) => const Home(title: 'APA'),
+        '/add': (context) => const AddField(),
+        '/fields': (context) => const MyFields(),
+        '/fly': (context) => const FlyDrone(droneName: 'DJI Mavic 3',),
+        '/settings': (context) => const Settings(),
```
Comment (paulmis) [inline src/apa_app/lib/main.dart:36]: Let's put the overview of app locations somewhere in the README or link the navigation diagram?

File: src/apa_app/lib/services/location_service.dart [hunk 4]
```diff
@@ -0,0 +1,29 @@
+import 'package:http/http.dart' as http;
+import 'dart:convert' as convert;
+import 'package:flutter_dotenv/flutter_dotenv.dart';
+
+class LocationService {
```
Comment (paulmis) [inline src/apa_app/lib/services/location_service.dart:5]: Can you describe what does the class do in a comment?

**Review by paulmis — COMMENTED** [review]
I would also call the main folder `mobile` or `frontend` rather than `apa_app`.  The application to control the drone would also be an `app`, so it's a bit ambiguous.

**Comment by paulmis** [comment]
Nicely done Nadine!

File: src/apa_app/lib/main.dart (hunk 3 above)
Comment (nadinekuo) [reply src/apa_app/lib/main.dart:36]: Yes good suggestion, I have added the current routes to the README - however, they are prone to modifications :)

File: src/frontend/lib/main.dart [hunk 5]
```diff
@@ -0,0 +1,95 @@
...
+  @override
+  Widget build(BuildContext context) {
+    return MaterialApp(
+      title: 'Autonomous Precision Agriculture using UAVs',
+      theme: ThemeData(
+        primarySwatch: Colors.lightGreen,
+      ),
+      initialRoute: '/',
+      routes: {
+        '/': (context) => const MainPage(),
```
Comment (DimanSUzunov) [inline src/frontend/lib/main.dart:30]: We could add a Page for invalid routes. Anything outside of the valid routes will go to this "Error" page.

File: src/frontend/lib/main.dart [hunk 6]
```diff
@@ -0,0 +1,95 @@
...
+
+  @override
+  Widget build(BuildContext context) {
+    return Scaffold(
+      // Ensures state of all children are preserved
+      body: IndexedStack(
+        index: currIndex,
+        children: screens,
+      ),
+      bottomNavigationBar: BottomNavigationBar(
```
Comment (DimanSUzunov) [inline src/frontend/lib/main.dart:69]: Do you have a defined component in another file for "BottomNavigationBar" ?

File: src/frontend/test/widget_test.dart [hunk 7]
```diff
@@ -0,0 +1,30 @@
+// This is a basic Flutter widget test.
```
Comment (DimanSUzunov) [inline src/frontend/test/widget_test.dart:1]: What is this file for?

**Review by DimanSUzunov — CHANGES_REQUESTED** [review]
I added some feedback for changes before approving. Check them out and let me know to review it again.

File: src/frontend/lib/main.dart (hunk 5 above)
Comment (nadinekuo) [reply src/frontend/lib/main.dart:30]: Sounds good, however shall we maybe add that later? (as it is not necessarily related to this issue) ;)

File: src/frontend/test/widget_test.dart (hunk 7 above)
Comment (nadinekuo) [reply src/frontend/test/widget_test.dart:1]: This test was in the project by default :) We may add our own tests later.

File: src/frontend/lib/main.dart (hunk 6 above)
Comment (nadinekuo) [reply src/frontend/lib/main.dart]: Initially I had kept it under `main.dart`, but it is a good suggestion so I just created a separate widget class

**Pull request merged**